### ISSUE TITLE
No error reported when nonlocal variable is annotated as global then used

### ISSUE THREAD

**Issue by narfanar** [microsoft/pylance-release#429]
## Environment data

-   Language Server version: 2020.9.6
-   OS and version: Windows Server 2019 Datacenter 1809
-   Python version (& distribution if applicable, e.g. Anaconda): Anaconda, Python 3.7.6 64bit

## Expected behaviour

Variables outside a nested function cannot be referenced unless annotated as `nonlocal`.

## Actual behaviour

`global` works just as fine. Even though no global variable has the name or type that shows up in the hint. It's seeing the nonlocal variable.

## Logs

Lots of this:
```
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
Background analysis message: getSemanticTokens
[BG(1)] parsing: Untitled-2 (2ms)
[BG(1)] binding: Untitled-2 (0ms)
Background analysis message: analyze
[BG(1)] analyzing: Untitled-2 ...
[BG(1)]   checking: Untitled-2 (0ms)
[BG(1)] analyzing: Untitled-2 (0ms)
```

## Code Snippet / Additional information

```python
global_foo : int = 23

def func():
    nonlocal_foo : str = ''
    def func_1():
        global nonlocal_foo
        nonlocal_foo = 23
```

![image](https://user-images.githubusercontent.com/5769148/94351329-f6efd680-0057-11eb-873c-cab7610c31a6.png)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of Pylance.

There is nothing technically wrong with your code sample. It runs without an exception in the Python interpreter, and Pylance should not generate any error for this example.

The bug here was specifically with the case where the `global` keyword referenced a symbol that was defined in a nonlocal (but also not global) scope.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.9.7, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202097-30-september-2020
